src/ui/pages/zcl_abapgit_gui_page_chg_pckg.clas.abap

```diff
@@ -0,0 +1,530 @@
+
+    update_sotr_package_assignment( lt_mapping ).
+
+    update_repo_persistence(
+      iv_old_package = iv_old_package
+      iv_new_package = iv_new_package ).
+
+    update_repo_checksums( lt_mapping ).
+
+    IF iv_remove_old = abap_true.
+      delete_packages( lt_tadir ).
+    ENDIF.
+
+    zcl_abapgit_repo_srv=>get_instance( )->init( ).
+
+  ENDMETHOD.
+
+
+  METHOD change_package_assignment.
+
+    DATA ls_tadir LIKE LINE OF it_tadir.
+
+    FIELD-SYMBOLS <ls_map> LIKE LINE OF it_mapping.
+
+    " TODO: Transportable packages (add to transport)
+    LOOP AT it_tadir INTO ls_tadir WHERE object <> 'DEVC' AND object <> 'NSPC'.
+      READ TABLE it_mapping ASSIGNING <ls_map> WITH KEY old_package = ls_tadir-devclass.
+      ASSERT sy-subrc = 0.
+
+      ls_tadir-devclass = <ls_map>-new_package.
+
+      CALL FUNCTION 'TR_TADIR_INTERFACE'
+        EXPORTING
+          wi_tadir_pgmid    = ls_tadir-pgmid
+          wi_tadir_object   = ls_tadir-object
+          wi_tadir_obj_name = ls_tadir-obj_name
+          wi_tadir_devclass = ls_tadir-devclass
+          wi_test_modus     = abap_false
+        EXCEPTIONS
+          OTHERS            = 1.
+      IF sy-subrc <> 0.
+        zcx_abapgit_exception=>raise_t100( ).
+      ENDIF.
+    ENDLOOP.
+
+  ENDMETHOD.
+
+
+  METHOD constructor.
+
+    super->constructor( ).
+    mo_repo = io_repo.
+    CREATE OBJECT mo_validation_log.
+    CREATE OBJECT mo_form_data.
+    mo_form = get_form_schema( ).
+    mo_form_util = zcl_abapgit_html_form_utils=>create( mo_form ).
+
+    IF mo_repo->get_dot_abapgit( )->get_folder_logic( ) <> zif_abapgit_dot_abapgit=>c_folder_logic-prefix.
+      zcx_abapgit_exception=>raise( 'Feature is only supported repositories with prefix folder logic' ).
+    ENDIF.
+
+    IF zcl_abapgit_factory=>get_cts_api( )->is_chrec_possible_for_package( mo_repo->get_package( ) ) = abap_true.
+      zcx_abapgit_exception=>raise( 'Feature is only supported local packages (no transport)' ).
+    ENDIF.
+
+  ENDMETHOD.
+
+
+  METHOD create.
+
+    DATA lo_component TYPE REF TO zcl_abapgit_gui_page_chg_pckg.
+
+    CREATE OBJECT lo_component
+      EXPORTING
+        io_repo = io_repo.
+
+    ri_page = zcl_abapgit_gui_page_hoc=>create(
+      iv_page_title      = 'Change Repository Package'
+      ii_child_component = lo_component ).
+
+  ENDMETHOD.
+
+
+  METHOD create_package_hierarchy.
+
+    DATA:
+      ls_tdevc  TYPE tdevc,
+      ls_tdevct TYPE tdevct,
+      ls_map    TYPE ty_map.
+
+    FIELD-SYMBOLS <ls_map> LIKE LINE OF it_mapping.
+
+    LOOP AT it_mapping ASSIGNING <ls_map>.
+      SELECT SINGLE * FROM tdevc INTO ls_tdevc WHERE devclass = <ls_map>-old_package.
+      ASSERT sy-subrc = 0.
+
+      ls_tdevc-devclass = <ls_map>-new_package.
+      ls_tdevc-as4user  = sy-uname.
+
+      READ TABLE it_mapping INTO ls_map WITH KEY old_package = ls_tdevc-parentcl.
+      IF sy-subrc = 0.
+        ls_tdevc-parentcl = ls_map-new_package.
+      ENDIF.
+
+      INSERT tdevc FROM ls_tdevc.
```

Comment:
huh?

Reply:
I dont expect any changes to SAP standard tables if it can be avoided

Reply:
⛷️ 😉 back next week

Reply:
done! sorry for the delay